WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -589,9 +589,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 }
                 break;
             case RequestCodes.REAUTHENTICATE:
-                if (resultCode == RESULT_CANCELED) {
```

Comment:
Yes. `RESULT_OK` happens if login completed, in which case this Activity just continues and starts the GCMRegistrationIntentService service.

If Login gets cancelled, app just returns to the main activity. Note, with the current implementation the login screen will be relaunched anyway, unless there's no connectivity.